JS/script.js

```diff
@@ -14,4 +14,122 @@ function hamFuntion() {
+    image: '',
+    technologies: ['Ruby on Rails', 'CSS', 'Javascript', 'HTML'],
+    live: '#',
+    source: '#',
+  },
+  {
+    name: 'Multi-Post Stories Gain+Glory',
+    description: 'Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s with the relea Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s with the relea',
+    image: '',
+    technologies: ['Ruby on Rails', 'CSS', 'Javascript', 'HTML'],
+    live: '#',
+    source: '#',
+  },
+];
+
+const cards = document.querySelector('.Cards');
+
+for (let i = 0; i < cardsInfo.length; i += 1) {
+  const element = document.createElement('section');
+  element.classList.remove(`WorksCard${i}`);
+  element.classList.add(`WorksCard${i + 1}`);
+  element.innerHTML = `
+    <div class="card1">
+    </div>
+    <div class="card2">
+      <h1>${cardsInfo[i].name}</h1>
+    </div>
+    <ul class="card3">
+      <li>${cardsInfo[i].technologies[0]}</li>
+      <li>${cardsInfo[i].technologies[1]}</li>
+      <li>${cardsInfo[i].technologies[2]}</li>
+      <li>${cardsInfo[i].technologies[3]}</li>
+    </ul>
+    <div class="card4">
+      <button class="CardBut">See project</button>
+    </div>`;
+  cards.appendChild(element);
+
+  const butSelect = element.querySelector('.CardBut');
+  butSelect.addEventListener('click', () => {
+    const newSec = document.createElement('section');
+    newSec.classList.add('pop');
+    document.querySelector('body').appendChild(newSec);
+    const popup = document.querySelector('.pop');
+    popup.innerHTML = `
+      <div class="popFlex">
+        <div class="closeButton">
+          <button class="popClose"><img srcset="Pics/Icons/xiconW.png 767w, Pics/Icons/xicon.png 1124w" alt="Icon"></button>
+        </div>
+      
+        <img srcset="Pics/imgPop.png 767w, Pics/imgPop2.png 1124w" alt="Photo">
+
+        <div class="items">
+
+          <h2 class="item1">Keeping track of hundreds of components</h2>
+
+          <ul class="item2">
+            <li>${cardsInfo[i].technologies[0]}</li>
+            <li>${cardsInfo[i].technologies[1]}</li>
+            <li>${cardsInfo[i].technologies[2]}</li>
+            <li>${cardsInfo[i].technologies[3]}</li>
+          </ul>
+
+          <p class="item3">${cardsInfo[i].description}
+          </p>
+
+          <div class="popButton item4">
+              <button>See source<img src="Pics/Icons/IconExp.png" alt="icon"></button>
+              <button>See live<img src="Pics/Icons/IconGit.png" alt="icon"></button>
+          </div>
+
+        </div>  
+      </div>
+
+    `;
+    const btnC = newSec.querySelector('.popClose');
+    btnC.addEventListener('click', () => {
+      newSec.remove();
+    });
+  });
```

Comment:
- [ ] Your code here appears to work but does not follow Javascript best practices. I suggest you refactor your code to make use of functions like the below 👇🏾 . This way your code becomes Modular, DRY, and SOLID. _**The code below is just a demo of what your own implementation could look like.**_  

```javascript
const cardsInfo = [
  {
    name: 'Multi-Post Stories Gain+Glory',
    description: 'Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s...',
    image: '',
    technologies: ['Ruby on Rails', 'CSS', 'Javascript', 'HTML'],
    live: '#',
    source: '#',
  },
  // Add other card objects...
];

// Function to create a card element from the card object
function createCardElement(cardInfo) {
  const template = document.getElementById('cardTemplate');
  const cardElement = template.content.firstElementChild.cloneNode(true);
  
  const titleElement = cardElement.querySelector('h1');
  const technologiesElement = cardElement.querySelector('.card3');
  
  titleElement.textContent = cardInfo.name;
  cardInfo.technologies.forEach(technology => {
    const liElement = document.createElement('li');
    liElement.textContent = technology;
    technologiesElement.appendChild(liElement);
  });
  
  const seeProjectButton = cardElement.querySelector('.CardBut');
  seeProjectButton.addEventListener('click', () => {
    createPopup(cardInfo);
  });

  return cardElement;
}

// Function to create the popup element for a card
function createPopup(cardInfo) {
  const template = document.getElementById('popupTemplate');
  const popupElement = template.content.firstElementChild.cloneNode(true);

  const closeButton = popupElement.querySelector('.popClose');
  closeButton.addEventListener('click', () => {
    popupElement.remove();
  });
  
  const titleElement = popupElement.querySelector('.item1');
  const technologiesElement = popupElement.querySelector('.item2');
  const descriptionElement = popupElement.querySelector('.item3');
  
  titleElement.textContent = cardInfo.name;
  cardInfo.technologies.forEach(technology => {
    const liElement = document.createElement('li');
    liElement.textContent = technology;
    technologiesElement.appendChild(liElement);
  });
  descriptionElement.textContent = cardInfo.description;

  // Set other popup data...

  document.querySelector('body').appendChild(popupElement);
}

// Function to render the cards on the page
function renderCards() {
  const cardsContainer = document.querySelector('.Cards');
  
  cardsInfo.forEach(cardInfo => {
    const cardElement = createCardElement(cardInfo);
    cardsContainer.appendChild(cardElement);
  });
}

// Call the renderCards function to display the cards
renderCards();
```

In this refactored code, I created two HTML templates, **cardTemplate** and **popupTemplate**, to define the structure of the card and popup elements. The JavaScript functions **createCardElement** and **createPopup** are responsible for creating the **card** and **popup** elements, respectively, by cloning the template elements and populating them with the corresponding data.

The **renderCards** function loops over the cardsInfo array, creates the card elements using the createCardElement function, and appends them to the .Cards container.

By separating the logic into smaller functions and using templates, I achieved better modularity, reusability, and maintainability of the code.

Reply:
Solved